### PR TITLE
Admin API: Create Tiers

### PR DESCRIPTION
**Pull request by Dannyps**
Enables the Admin API to be used for the creation of Tiers.

### CODE REVIEW

**Comment by tnotheis** [comment]
This branch needs to be synced with main

**Comment by Dannyps** [comment]
@tnotheis, yesterday @daniel-almeida-konkconsulting mentioned you may prefer this PR to target the parent branch of the branch to be merged, which is `feature/DEV-1808-admin-api-identities-list` (#46).
Should I change the base, or sync this wih `main`?

**Comment by tnotheis** [comment]
@Dannyps Sorry, my bad. I meant the base branch, of course.

**Review by tnotheis — CHANGES_REQUESTED** [review]
Shouldn't the target branch of this PR be the `feature/create-tier` branch?

**Comment by Dannyps** [comment]
I believe I've fixed this as best I can. The `feature/create-tier` branch does not have anything related to the Admin.API.

The way I see it, we can either bring the Admin API to it via this PR, or we can merge main to it, strip the Admin.API commits from this PR, and merge.

File: Modules/Devices/src/Devices.Application/Infrastructure/Persistence/Repository/IIdentityRepository.cs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+﻿using Backbone.Modules.Devices.Domain.Entities;
+using Enmeshed.BuildingBlocks.Application.Abstractions.Infrastructure.Persistence.Database;
+using Enmeshed.BuildingBlocks.Application.Pagination;
+
+namespace Backbone.Modules.Devices.Application.Infrastructure.Persistence.Repository;
+
+public interface IIdentityRepository
```
Comment (tnotheis) [inline Modules/Devices/src/Devices.Application/Infrastructure/Persistence/Repository/IIdentityRepository.cs:7]:
```suggestion
public interface IIdentitiesRepository
```

File: Modules/Devices/src/Devices.Infrastructure/Persistence/Repository/IdentityRepository.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace Backbone.Modules.Devices.Infrastructure.Persistence.Repository;
+public class IdentityRepository : IIdentityRepository
+{
+    private readonly DbSet<Identity> _identityDbSet;
+    private readonly IQueryable<Identity> _readonlyIdentityDbSet;
+
+    public IdentityRepository(DevicesDbContext dbContext)
+    {
+        _identityDbSet = dbContext.Identities;
```
Comment (tnotheis) [inline Modules/Devices/src/Devices.Infrastructure/Persistence/Repository/IdentityRepository.cs:17]: Remove this as long as we don't need it, please.

File: Modules/Devices/test/Devices.API.Tests.Integration/Models/IdentityResponse.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿using Devices.API.Tests.Integration.Models;
+using Newtonsoft.Json;
+
+namespace Challenges.API.Tests.Integration.Models;
```
Comment (tnotheis) [inline Modules/Devices/test/Devices.API.Tests.Integration/Models/IdentityResponse.cs:4]: Wrong Namespace

**Review by tnotheis — CHANGES_REQUESTED** [review]
There are no integration tests for the route.

**Review by tnotheis — APPROVED** [review]

**Pull request merged**